overlay.d/05core/usr/lib/dracut/modules.d/35coreos-ignition/coreos-disk-contains-fs.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+# checks whether `disk` contains filesystem labeled `label`
+set -euo pipefail
+
+disk=$1
+label=$2
+
+# during execution of udev rules on disks 'lsblk' returns empty fields
+for pt in /sys/block/$disk/*; do
+    name=$(basename $pt)
+    if [[ "$name" =~ "$disk" ]]; then
```

Comment:
Was curious how libblk did this, and found https://github.com/util-linux/util-linux/blob/177c9ec7c7aa0f49413f4cb5f347edf721d5a5ba/lib/sysfs.c#L203-L243, which isn't far from what you have here.

How about something like this to tighten this up:

```suggestion
    if [[ "$name" =~ ${disk}p?[[:digit:]] ]] && [ -e "/sys/block/$disk/$name/start" ]; then
```

?